catalogbuilder/intakebuilder/gfdlcrawler.py

```diff
@@ -114,12 +117,25 @@ def crawlLocal(projectdir, dictFilter,dictFilterIgnore,logger,configyaml,slow):
                # todo do the reverse if slow is on. Open file no matter what and populate dictionary values and if there is something missed out
                # we can scan filenames or config etc 
                #here, we will see if there are missing header values and compare with file attributes if slow option is turned on
-               if (slow == True) & (bool(dictInfo) == True) :
-                    print("Slow option turned on.. lets open some files using xarray and lookup atts",filename)
-                    #todo we could look at var attributes, but right now we stick to those that are necessary. scope to extend this easily to missngcols or if header info is not in config yaml 
-                    if "standard_name" in missingcols: 
+               if (slow == True) & (bool(dictInfo) == True):
+                    print("Slow option turned on.. lets open some files using xarray and lookup atts")
+                    
+                    #todo we could look at var attributes, but right now we stick to those that are necessary. scope to extend this easily to missngcols or if header info is not in config yaml
+                    if "standard_name" in missingcols:
+
                         dictInfo["standard_name"] = "na"
-                        getinfo.getInfoFromVarAtts(dictInfo["path"],dictInfo["variable_id"],dictInfo)
+
+                        #Check if we've come across a similar dataset
+                        qualities=[dictInfo["variable_id"],dictInfo["realm"]]
+                        for standard_name,quality_list in unique_datasets.items():
+                            if quality_list == qualities:
+                                dictInfo["standard_name"]=standard_name
+
+                        if dictInfo["standard_name"] == "na":
+                            print("Retrieving standard_name from ", filename)
+                            getinfo.getInfoFromVarAtts(dictInfo["path"],dictInfo["variable_id"],dictInfo)
+                            unique_datasets.update({ dictInfo["standard_name"] : qualities})
```

Comment:
@Ciheim Headed in the right direction. Switch the keys and values in the dictionary though. The standard name is not a unique key across different "qualities". So using some form of "qualities" as key, and assigning standard name as its values makes most sense. 

You may also explore updating the dictInfo["standard_name"] as part of this routine. That way, you don't need a post-processing step of the csv like I did with the look-up table.

Reply:
@Ciheim your comments from today's sprint makes sense. dictInfo is being reset so the duplicate keys should not be an issue I hope. Your verification as you test this today will reveal more. thanks!
https://github.com/NOAA-GFDL/CatalogBuilder/blob/54794ad9c1361e72bc8a00a9840361c6f641c8a9/catalogbuilder/intakebuilder/gfdlcrawler.py#L89